### ISSUE TITLE
NexT v7.x Roadmap

### ISSUE THREAD

**Issue by ivan-nginx** [theme-next/hexo-theme-next#67]
## Requested features:

- [ ] Fluidbox or mediumZoom? ([#394](https://github.com/iissnan/hexo-theme-next/issues/394), #627)
- [ ] ~~Filters on archives/cats/tags ([#424](https://github.com/iissnan/hexo-theme-next/issues/424))~~
- [ ] PJAX support ([#918](https://github.com/iissnan/hexo-theme-next/issues/918))
- [x] Related popular posts ([#1130](https://github.com/iissnan/hexo-theme-next/issues/1130), added in #109)
- [ ] Night Mode on NexT ([#1871](https://github.com/iissnan/hexo-theme-next/issues/1871))
- [ ] Lazy loading for all possible contents (infinite loading)? ([#1914](https://github.com/iissnan/hexo-theme-next/issues/1914))
- [x] Secondary menu? ([#1924](https://github.com/iissnan/hexo-theme-next/issues/1924), added in #156)
- [x] Overlength toc cannot show full content in `wrap: false` mode ([#1980](https://github.com/iissnan/hexo-theme-next/issues/1980), #176)
- [ ] Variable templates on schemes with misc colors (not only black-white)
- [ ] Increase site loading speed in Google Speed Insights.
- [x] Option `text-align` on mobile (not only on mobile?) for all schemes (#73, added in #682)
- [ ] ~~Refactor headers in all schemes & add `post.*` options to centrilized H1-H3 titles.~~
- [x] Display all level of TOC instead of being activated. (#307, added in #679)
- [ ] Changing color templates with JS directly and save to cache.
- [x] Improve time visible for `symbols-count-time` - if less than 60 mins, show only mins ((https://github.com/theme-next/hexo-symbols-count-time/issues/11), added in #233)
- [x] Sidebar width in settings (addition to #103, added in #111, refactored in #339)
- [ ] Sidebar on mobile for all schemes (not only for Muse | Mist)
- [x] Sidebar to the left in Mist (#164). Need to add to Muse scheme too. ([#803](https://github.com/iissnan/hexo-theme-next/issues/803))
- [x] Enable/Disable global animation (to make velocity external, #113)
- [x] Clipboard or `select all` button for code blocks. (added in #234)
- [x] Custom JS files. (Using theme-inject, #868)
- [ ] ~~Custom page layout (gallery and product page) (#144)~~
- [ ] ~~Album or PhotoWall page (#295)~~
- [x] 4k resolution adaptation (#150, added in #339)
- [ ] `spoiler` tag (#200)
- [x] Make font customization more flexible, use `em` instead of `px` as default font-size unit (#249, #726)
- [x] Modify `exturl` into internal filter. (added in #463)
- [ ] ~~Option for 100% or not for iframe (#316)~~
- [x] Custom 404 page (#389, Docs added)
- [ ] ~~Add post sticky feature? (#447)~~ Adding https://github.com/stevenjoezhang/hexo-generator-indexed to the docs is enough
- [x] Option to Hide/Show back2top button. (#482, added in #657)
- [ ] ~~Function for detect browsers, similar as already exists function to detect UA.~~

### Third-party:

- [x] Support mermaid (using tag)? (#347, added in #649)
- [x] Likely support (#491, added in #678)
- [x] Add Charta (#577, added in #681)
- [x] disqusJS comment system (#586, added in #705)
- [ ] ~~Google/Yandex maps tags.~~ Too many arguments required for this tag. Using https://www.maps.ie/create-google-map/ to create an iframe would be better

## Known bugs:

- [x] Rename `Words count in article` to `Symbols count in article` (fixed in 9d46733e2f9b776dfe1bab5e65be0937dd4ca5b8)
- [x] Categories bug? ([#300](https://github.com/iissnan/hexo-theme-next/issues/300), was solved)
- [x] Sidebar bug? ([#442](https://github.com/iissnan/hexo-theme-next/issues/442))
- [x] Mist scheme bug? ([#459](https://github.com/iissnan/hexo-theme-next/issues/459), fixed in https://github.com/iissnan/hexo-theme-next/commit/1985b7ecced0df4cfe2d9cf6337b40040d1ebf60#diff-ddaaf61bc40a2d092db6ddb0715dbde0)
- [x] Make table scrollable ([#1582](https://github.com/iissnan/hexo-theme-next/issues/1582), fixed in #598)
- [x] W3 validator Baidu errors ([#1646](https://github.com/iissnan/hexo-theme-next/issues/1646))
- [x] Algolia still in bug? ([#1764](https://github.com/iissnan/hexo-theme-next/issues/1764), fixed in #89)
- [x] Option to disable wechat/alipay/bitcoin reward animation ([#1856](https://github.com/iissnan/hexo-theme-next/issues/1856))
- [x] Tagcloud generation limit ([#1908](https://github.com/iissnan/hexo-theme-next/issues/1908))
- [x] Leancloud visitor counter security bug (added in #25)
- [x] Refactor font styles to global variables for all schemes (#38)
- [x] Refactor footer alignment for all schemes (#48)
- [x] Fix & improve styles in codeblocks titles (#100, fix and feature added in #194)
- [x] Move `busuanzi_count` languages values from config to laguages files (#110)
- [x] Paddings in `auto_excerpt: enable: true` with `mobile_layout_economy` option (#181, fixed in #534)
- [x] Update `hexo-generator-searchdb` (#190, https://github.com/theme-next/hexo-generator-searchdb/pull/5)

## Miscellaneous:

- [x] Replace SWIG? (#33, [#1892](https://github.com/iissnan/hexo-theme-next/issues/1892))
 Already replaced `swig` by `swig-templates` in [this pull](https://github.com/hexojs/hexo/pull/2949) and released in [Hexo v3.5.0](https://github.com/hexojs/hexo/releases/tag/3.5.0)
- [ ] Script for auto-increase version before initial release into tag.
- [ ] AppVeyor & Build automatization scripts.
- [ ] Separate schemes on modules/submodules.
- [ ] Separate styles between schemes (exclude all possible styles rewrites)
- [ ] Move all possible docs with features from config to site with docs.
- [x] Custom language file support (`_custom.yml`) (added in #253).
- [x] Document for group-images usage (move from `theme/next/scripts/tags/group-pictures.js` to theme-next.org, [#1073](https://github.com/iissnan/hexo-theme-next/issues/1073))
- [x] Remove `exturl` tag? (`theme/next/scripts/tags/exturl.js`, warnings added in https://github.com/theme-next/hexo-theme-next/pull/766)
- [x] `.deploy_git` & `.nojekyll` ([#1214](https://github.com/iissnan/hexo-theme-next/issues/1214#issuecomment-260847236))
- [ ] Language switcher by flags? ([#1680](https://github.com/iissnan/hexo-theme-next/issues/1680))
- [ ] Refactor all separated transactions on one global transaction.
- [ ] Refactor source/css/_common/components/post/post-type.styl
- [ ] ~~Refactor main font name and compare compatability for different languages & browsers.~~
- [ ] ~~ForkAwesome / FontAwesome? (#94, #474 and #505)~~
- [ ] Console reminder if new version released [autoupdate option] (#318)
- [ ] ~~NexT 3rd-party modules as submodules? (#413, #564 and #663)~~
- [ ] Refactoring Mathjax? (#257, #645, #658) Transform latex expression to svg in build time?
- [ ] Archives & Categories & Tags counting for plural counting in languages.

***

Please feel free to leave your thought in comments (or edit this post directly if u have permissions).

**Comment by tsanie** [comment]
pjax is really cool, I think its priority could be set to high. 😃

**Comment by ivan-nginx** [comment]
There is no matter order in lists. Just all finded and remembered features/bugs for now. This list can be also increased in any time.

**Comment by ApolloZhu** [comment]
Maybe supporting sidebar on mobile for all schemes?

**Comment by ivan-nginx** [comment]
@ApolloZhu added.

**Comment by maple3142** [comment]
I think instantclick works like pjax.(And it is easy to add)
But add it will break many thing.

**Comment by GARENFEATHER** [comment]
wow, so many new features I appreciate, anyone that has upgraded?

**Comment by ivan-nginx** [comment]
This is roadmap, not actually implemented features.

**Comment by tiny1990** [comment]
Code highlighting plugin is not pretty.

**Comment by BillYang2016** [comment]
Clipboard or "select all" button for code blocks.

**Comment by ivan-nginx** [comment]
@BillYang2016 added.

**Comment by maple3142** [comment]
How do you think about upgrade to jquery 3?

**Comment by ivan-nginx** [comment]
Yeah, seems good. And if variant, I think about remove out jQuery totally.

**Comment by maple3142** [comment]
I think remote jQuery totally is a little bit impossible, because too many plugin depends on it.
Like fancybox, lazyload.

**Comment by ivan-nginx** [comment]
Set as dependencies of course, not possible?
Anyway, i think there is some versions of «light» types jQuery. We can try to use more lightweight lib. I see what some CN version was like this, but i don't und lang. U can try to researh it.

**Comment by maple3142** [comment]
About pjax, here has an userscript example implement a really basic approach to pjax.
https://gist.github.com/maple3142/d9c752cebfdd92a7edd1e59ea2cd2f6a
It works on https://almostover.ru/ and https://rainylog.com/

But it usually break some jquery plugin like scrollspy, velocity...

**Comment by LEAFERx** [comment]
Requesting for announcement feature, like a lighter post which doesn't have a single page but only on the main page to put some notices/info/states/updates/mood sharings.

**Comment by ivan-nginx** [comment]
@LEAFERx i a little bit don't understand what u mean, but u can add it to roadmap by yourself.

**Comment by maple3142** [comment]
How do you think about static MathJax?
Transform latex expression to svg in build time.

Should it be theme feature or a independent hexo plugin?

Example: https://blog.maple3142.net/about/

**Comment by ivan-nginx** [comment]
@maple3142 cool! It seems this will load faster then Math* plugins!

> Should it be theme feature or a independent hexo plugin?

I think this no matter, but also better will to separate this feature with external module or plugin. Do with your own opinion and for better:

* Speed
* Usability
* Compatability.

**Comment by baoang** [comment]
Will Next provide style kit for jekyll and hugo in the coming v7?

**Comment by maple3142** [comment]
@baoang I think it is impossible, or it needs too much change.
Why not use hexo, or use other jekyll/hugo theme?
If you need a jekyll theme, [Simpleyyt/jekyll-theme-next](https://github.com/Simpleyyt/jekyll-theme-next) is a port version of NexT theme.

**Comment by ivan-nginx** [comment]
@baoang @maple3142 maybe we can to ask developers like @Simpleyyt to join to main NexT organization and resume maintenance here?

**Comment by ChungZH** [comment]
I think the codeblock should have a variety of styles (but not highlight_theme)

**Comment by stevenjoezhang** [comment]
Other known issues:
- about .gitignore (https://github.com/theme-next/hexo-theme-next/issues/564 and https://github.com/theme-next/hexo-theme-next/issues/413)
- function `getContentVisibilityHeight` in `source/js/src/utils.js`, `$('#content').height();` or `$('.container').height()` (#48)

**Comment by jodumont** [comment]
I wonder when next team plan to move to bootstrap4 ?

**Comment by jodumont** [comment]
@stevenjoezhang thanks for the hint

also I really love next theme it perform very well and it is outstanding
but when I compare it with [this jekyll theme](https://idratherbewriting.com/documentation-theme-jekyll/index.html) and I see how visually the theme and menu are clear; I'm kind of jealous.

### Make the sidebar optional ?

also would be nice to being able to render page without the sidebar.

**Comment by ivan-nginx** [comment]
> also would be nice to being able to render page without the sidebar.

```md
title: ...
sidebar: false
```

**Comment by jodumont** [comment]
@stevenjoezhang @ivan-nginx Wow! now I scared to ask if is this theme is able to make coffee ;)

**Comment by JiangTJ** [comment]
@ivan-nginx @stevenjoezhang Maybe we can manage issues in projects, not here.

**Comment by stevenjoezhang** [comment]
See: https://github.com/theme-next/hexo-theme-next/issues/809
